--- libvmime-0.9.2/configure.in.libgcrypt~	2010-12-13 10:26:04.000000000 +0100
+++ libvmime-0.9.2/configure.in	2012-01-26 16:35:41.431713559 +0100
@@ -317,6 +317,16 @@ else
 	VMIME_HAVE_MESSAGING_FEATURES=0
 fi
 
+	AC_CHECK_HEADER(gcrypt.h,
+		AC_CHECK_LIB(gcrypt, gcry_md_open,
+			[have_gcrypt=yes AC_SUBST(GCRYPT_LIBS, -lgcrypt)],
+			 have_gcrypt=no),
+		have_gcrypt=no)
+
+	if test "x$have_gcrypt" != "xyes"; then
+		AC_MSG_ERROR(can't find an usable version of Libgcrypt)
+	fi
+
 # ** SASL
 
 AC_ARG_ENABLE(sasl,
@@ -618,7 +628,7 @@ AC_SUBST(PKGCONFIG_CFLAGS)
 AC_SUBST(PKGCONFIG_LIBS)
 
 EXTRA_CFLAGS="$EXTRA_CFLAGS -D_REENTRANT=1 -D_THREAD_SAFE=1 $LIBGNUTLS_CFLAGS"
-EXTRA_LIBS="$GSASL_LIBS $LIBGNUTLS_LIBS"
+EXTRA_LIBS="$GSASL_LIBS $LIBGNUTLS_LIBS $GCRYPT_LIBS"
 
 CFLAGS=""
 CXXFLAGS=""
