diff -urN libvmime-0.9.2/src/net/tls/TLSSession.cpp libvmime-0.9.2-patched/src/net/tls/TLSSession.cpp
--- libvmime-0.9.2/src/net/tls/TLSSession.cpp	2010-09-14 00:30:32.000000000 +1100
+++ libvmime-0.9.2-patched/src/net/tls/TLSSession.cpp	2012-11-17 04:44:27.738134843 +1100
@@ -22,7 +22,6 @@
 //
 
 #include <gnutls/gnutls.h>
-#include <gnutls/extra.h>
 
 #include "vmime/config.hpp"
 
@@ -298,10 +297,7 @@
 	ERROR(GNUTLS_E_FILE_ERROR)
 	ERROR(GNUTLS_E_TOO_MANY_EMPTY_PACKETS)
 	ERROR(GNUTLS_E_UNKNOWN_PK_ALGORITHM)
-	ERROR(GNUTLS_E_INIT_LIBEXTRA)
-	ERROR(GNUTLS_E_LIBRARY_VERSION_MISMATCH)
 	ERROR(GNUTLS_E_NO_TEMPORARY_RSA_PARAMS)
-	ERROR(GNUTLS_E_LZO_INIT_FAILED)
 	ERROR(GNUTLS_E_NO_COMPRESSION_ALGORITHMS)
 	ERROR(GNUTLS_E_NO_CIPHER_SUITES)
 	ERROR(GNUTLS_E_OPENPGP_GETKEY_FAILED)
