diff -Nurb libvmime-0.8.1.orig/src/charset.cpp libvmime-0.8.1.patched/src/charset.cpp
--- libvmime-0.8.1.orig/src/charset.cpp	2006-10-11 16:52:40.000000000 +0200
+++ libvmime-0.8.1.patched/src/charset.cpp	2007-12-12 23:16:38.167888653 +0100
@@ -44,6 +44,9 @@
 charset::charset(const string& name)
 	: m_name(name)
 {
+	// if we receive this rfc-1642 valid MIME charset, convert it to something usefull for iconv
+	if (utility::stringUtils::isStringEqualNoCase(m_name, "unicode-1-1-utf-7"))
+		m_name = "utf-7";
 }
 
 
@@ -59,6 +62,10 @@
 	m_name = utility::stringUtils::trim
 		(string(buffer.begin() + position, buffer.begin() + end));
 
+	// if we parsed this rfc-1642 valid MIME charset, convert it to something usefull for iconv
+	if (utility::stringUtils::isStringEqualNoCase(m_name, "unicode-1-1-utf-7"))
+		m_name = "utf-7";
+
 	setParsedBounds(position, end);
 
 	if (newPosition)
