Index: svn/src/body.cpp
===================================================================
--- svn/src/body.cpp	(revision 593)
+++ svn/src/body.cpp	(working copy)
@@ -197,6 +197,10 @@
 			{
 				ref <bodyPart> part = vmime::create <bodyPart>();
 
+				// end before start may happen on empty bodyparts (directly successive boundaries without even a line-break)
+				if (partEnd < partStart)
+					std::swap(partStart, partEnd);
+
 				part->parse(buffer, partStart, partEnd, NULL);
 				part->m_parent = m_part;
 
